### PR TITLE
docs: add committer retirement guidelines

### PR DESCRIPTION
**Pull request by mspiekermann**
## What this PR changes/adds

Adding documentation to the committer retirement process and info about committers role in the project.

## Why it does that

We need to document how we will proceed if committers become inactive for any reason.

### CODE REVIEW

File: COMMITTERS.md [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+- Engage in the Intellectual Property Due Diligence Process;
+- Vote in committer and project lead elections;
+- Engage in the project planning process; and
+- Otherwise represent the interests of the open source project.
+
+For Eclipse projects (and the open source world in general), committers are the ones who hold the keys. Committers are either appointed at the time of project creation or elected by the existing project team.
+
+## Inactive Committers
+
+It's inevitable, but there are times when someone shifts focus, changes jobs, or retires from a particular area of the project (for a period of time). These people may be experts in certain areas of the codebase or reference persons for certain topics but can no longer devote the time necessary to take on the responsibilities of a committer role.  However, being a Committer within an Eclipse Foundation project comes with an elevated set of permissions. These capabilities should not be used by those that are not familiar with the current state of the EDC project.
```
Comment (juliapampus) [inline COMMITTERS.md:15]:
```suggestion
It's inevitable, but there are times when someone shifts focus, changes jobs, or retires from a particular area of the project (for a period of time). These people may be experts in certain areas of the codebase or reference persons for certain topics but can no longer devote the time necessary to take on the responsibilities of a committer role. However, being a Committer within an Eclipse Foundation project comes with an elevated set of permissions. These capabilities should not be used by those that are not familiar with the current state of the EDC project.
```

File: COMMITTERS.md [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+
+From time to time, it is necessary to prune inactive folks. A core principle in maintaining a healthy community is encouraging active participation. Those listed as a Committer of the project have a higher activity requirement, as they directly impact the ability of others to contribute. Therefore, members with an extended period away from the project with no activity will be retired from their role of Committer
+in the EDC and will be required to go through the meritocratic process again after re-familiarizing themselves with the current state. Committers, that can no longer devote the time are kindly asked to follow the retirement process of the Eclipse Foundation.
+
+According to the EF rules, before retiring a committer, the project’s community will be informed of the change and the Committer must be given a chance to defend retaining their status via the project’s dev-list.
```
Comment (juliapampus) [inline COMMITTERS.md:20]:
```suggestion
According to the EF rules, before retiring a Committer, the project’s community will be informed of the change and the Committer must be given a chance to defend retaining their status via the project’s dev-list.
```

stay consistent

File: COMMITTERS.md [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+
+To honor the contributions, retired committers are listed as Historic Committers on a project’s Who’s Involved page. When a committer returns to being more active in that area, they may be promoted back on the decision of the committers' committee. However, after an extended period away from the project with no activity those would need demonstrably have to re-familiarize themselves with the current state before being able to contribute effectively.
```
Comment (juliapampus) [inline COMMITTERS.md:22]:
```suggestion
To honor the contributions, retired Committers are listed as Historic Committers on the project’s [Who’s Involved page](https://projects.eclipse.org/projects/technology.edc/who). When a Committer returns to being more active in that area, they may be promoted back on the decision of the Committers' committee. However, after an extended period away from the project with no activity those would need demonstrably have to re-familiarize themselves with the current state before being able to contribute effectively.
```

File: docs/decision-records/2023-07-06-committer-retirement/README.md
Comment (juliapampus) [inline docs/decision-records/2023-07-06-committer-retirement/README.md:1]: Please also add a link to this document to the `decisions-records/README.md`.

File: docs/decision-records/2023-07-06-committer-retirement/README.md [hunk 4]
```diff
@@ -0,0 +1,22 @@
+# Adding guidelines for committers retirement
+
+## Decision
+[Committer of the EDC](https://projects.eclipse.org/projects/technology.edc/who) project have a higher activity requirement, as they directly impact the ability of others to contribute. Therefore, committers with an extended period away from the project with no activity will be retired from their role of _"committer"_ in the EDC and will be required to go through the meritocratic process again after re-familiarizing themselves with the current state.
```
Comment (juliapampus) [inline docs/decision-records/2023-07-06-committer-retirement/README.md:4]: Why font highlighting and quotationmarks? Italic should be sufficient.

**Review by juliapampus — CHANGES_REQUESTED** [review]

**Review by juliapampus — APPROVED** [review]

**Review by ronjaquensel — APPROVED** [review]

File: docs/decision-records/2023-07-06-committer-retirement/README.md [hunk 5]
```diff
@@ -0,0 +1,22 @@
+# Adding guidelines for Committers retirement
+
+## Decision
+[Committer of the EDC](https://projects.eclipse.org/projects/technology.edc/who) project have a higher activity requirement, as they directly impact the ability of others to contribute. Therefore, *Committers* with an extended period away from the project with no activity will be retired from their role of *Committer* in the EDC and will be required to go through the meritocratic process again after re-familiarizing themselves with the current state.
+
+Inactive *Committers* are defined as members of the EDC *Committers* with no contributions across any repository within 6 months (according to EF’s suggestion), or recognizable non-code contributions to the project. *Committers* are required to show up in *Committer-Meetings*.
```
Comment (jimmarino) [inline docs/decision-records/2023-07-06-committer-retirement/README.md:6]: If we say committers are required to show up to meetings, we should define what that is. I also don't think it should be an onerous requirement. Maybe we should rephrase this as:

 "Committers are expected to regularly attend commuter meetings"

**Review by jimmarino — APPROVED** [review]

**Review by alexandrudanciu — APPROVED** [review]
Please consider some minor improvement suggestions.

**Review by wolf4ood — APPROVED** [review]
